zapcore/field.go

```diff
@@ -199,3 +199,25 @@ func addFields(enc ObjectEncoder, fields []Field) {
 		fields[i].AddTo(enc)
 	}
 }
+
+func encodeStringer(key string, stringer interface{}, enc ObjectEncoder) {
+	defer catchPanic(key, enc)
```

Comment:
ok with inlining `catchPanic`, especially since we'll need to modify the returned error here (if we make `encodeStringer` return `error`:

```go
func encodeStringer(key string, stringer interface{}, enc ObjectEncoder) (retError error) {
  defer func() 
    if err := recover(); err != nil {
      // massage error
      retError = err
    }
  }()

  enc.AddString(key, stringer.(fmt.Stringer).String())
```